### PR TITLE
BFD-351: EOB resource and PDE Profile for BFD V2

### PR DESCRIPTION
**Pull request by jzulim**
### Change Details
- Adds an FHIR R4 and CARIN 'mostly' compliant ExplainationOfBenefit resource. 
- Adds the PDE claim type to ExplainationOfBenefit resource

- TODO: Add unit and integration tests. - [BFD-533](https://jira.cms.gov/browse/BFD-533)
- TODO: Resolve conformance issues - [BFD-583](https://jira.cms.gov/browse/BFD-583)

### Acceptance Validation

Note: No UT or IT's were created for this code yet. We have this as a TODO (see above)
Note: The FHIR conformance will need to be resolved. 

### Feedback Requested

- Functionality looks correct
- Naming is consistent with previous patterns and make sense. 
- Comments are where they need to be, informing reader of the 'what' and not the 'how.'
- Area for improving efficiency or reducing repeat code.

### External References

- [BFD-351](https://jira.cms.gov/browse/BFD-351)

### Security Implications

- [ x ] new software dependencies

- We are making use of org.hl7.fhir.r4.model.ExplanationOfBenefit;

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Review by cbrunefearless — DISMISSED** [review]
lgtm

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/V2Server.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ private void configureServerInfoMetadata() {
 
     // Lightly customize the capability provider to set publisher name.
     ServerCapabilityStatementProvider capabilityStatementProvider =
-        new ServerCapabilityStatementProvider(this);
```
Comment (jzulim) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/V2Server.java:69]: Arg constructor is deprecated. Use no args construct.  
ref: https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/hapi/rest/server/ServerCapabilityStatementProvider.html

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+    Matcher eobIdMatcher = EOB_ID_PATTERN.matcher(eobIdText);
+    if (!eobIdMatcher.matches())
+      throw new IllegalArgumentException("Unsupported ID pattern: " + eobIdText);
+
+    String eobIdTypeText = eobIdMatcher.group(1);
+    Optional<ClaimType> eobIdType = ClaimType.parse(eobIdTypeText);
+    if (!eobIdType.isPresent()) throw new ResourceNotFoundException(eobId);
+    String eobIdClaimIdText = eobIdMatcher.group(2);
+
+    Operation operation = new Operation(Operation.Endpoint.V1_EOB);
```
Comment (jzulim) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java:140]: Should be V2_EOB

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/ClaimType.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import java.util.function.Function;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.metamodel.PluralAttribute;
+import javax.persistence.metamodel.SingularAttribute;
+import org.hl7.fhir.r4.model.ExplanationOfBenefit;
+
+/**
+ * Enumerates the various Blue Button claim types that are supported by {@link
```
Comment (jzulim) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/ClaimType.java:22]: BFD

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2.java [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+
+    if (!(claim instanceof PartDEvent)) throw new BadCodeMonkeyException();
+    ExplanationOfBenefit eob = transformClaim((PartDEvent) claim);
+
+    timer.stop();
+    return eob;
+  }
+
+  /**
+   * @param claimGroup the CCW {@link PartDEvent} to transform
```
Comment (jzulim) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2.java:47]: @jarwilliams1971 Here's the PDE Mapping

**Review by jarwilliams1971 — COMMENTED** [review]
From what I can see it looks good from a layman's perspective

**Review by Dranith — APPROVED** [review]
I've gone through the V2 EOB transformer and related code pretty closely at this point.  Besides some minor questions that can be cleared up later, the contents of this look good.

**Review by cbrunefearless — APPROVED** [review]
LGTM

**Pull request merged**